libs/ledger-live-common/src/exchange/swap/getExchangeRates.ts

```diff
@@ -40,19 +37,64 @@ const getExchangeRates: GetExchangeRates = async (
   const amountFrom = transaction.amount;
   const tenPowMagnitude = new BigNumber(10).pow(unitFrom.magnitude);
   const apiAmount = new BigNumber(amountFrom).div(tenPowMagnitude);
+
+  const providerList =  providers.filter(item => {
+    const index = item.pairs.findIndex(pair => pair.from === from && pair.to === to);
+    return index >= -1;
+  }).map(item => item.provider);
+
+  const decentralizedSwapAvailable = () => {
+    const {
+      fromAccount: sourceAccount,
+      toAccount: targetAccount,
+      fromParentAccount: sourceParentAccount,
+      toParentAccount: targetParentAccount
+    } = exchange;
+
+    if (sourceAccount && targetAccount) {
+      const sourceMainAccount = getMainAccount(sourceAccount, sourceParentAccount);
+      const targetMainAccount = getMainAccount(targetAccount, targetParentAccount);
+
+      if (
+        targetMainAccount.currency.family === "ethereum" &&
+        sourceMainAccount.currency.id === targetMainAccount.currency.id
+      ) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  // TODO Uncomment if and update item attribute with DEX type definition
+
+  if(!providerList.length && decentralizedSwapAvailable()) {
+  //   const dexProviders = ["paraswap", "oneinch"]
```

Comment:
Just trying to understand what is going to happen when a requested currency pair isn't supported and `decentralizedSwapAvailable` returns true. I was assuming that we will also want to return the other CEX/DEX rates that were supported, in which case the early return here wouldn't be what we'd need. 

I was wondering do we need to add to the returned data from `res.data.map` [block](https://github.com/LedgerHQ/ledger-live/pull/1881/files#diff-7a950ca72c6e2330740af42542b2796bcb78cd7455cec031949dfd03f370d037R100) and append to it these placeholder quotes. Would adding something like the following work:

```
const results = res.data.map(existingCode)
const dexProviders = ["paraswap", "oneinch"]
for (const dexProvider in dexProviders) {
  if (!providersList.includes(dexProvider) && decentralizedSwapAvailable) {
      results.push({
        magnitudeAwareRate: undefined,
         provider,
         rate: undefined,
         rateId: undefined,
         toAmount: undefined,
         tradeMethod: "float",
         payoutNetworkFees: undefined
       })
  }
}
```

Reply:
Updated. You are right!